cdap-security/src/main/java/io/cdap/cdap/security/auth/TinkCipher.java

```diff
@@ -66,6 +73,15 @@ public TinkCipher(SConfiguration sConf) throws CipherException {
     }
   }
 
+  public static synchronized TinkCipher getInstance(SConfiguration sConf) throws CipherException {
```

Comment:
What kind of latency we allow for this call? Using synchronized across multiple threads frequently can hinder the performance quite a lot

Reply:
A quick benchmark of encryption and decryption 1k string total 500K times across either single or 10 threads before and after.  Ran a few times. 

Before PR: 
1 thread: 43 seconds 

After PR
1 thread: 27 seconds
10 threads: 7 seconds per thread (aggregated 75 seconds) 

Caveats: 
* Run on macbook. Not done in dedicated machine. 
* Run in intellji

Reply:
How about 10 threads before PR??

Reply:
Before PR: 
10 threads: ~11 seconds per thread (aggregated 113) 

After PR: 
10 threads: ~7 seconds per thread (aggregated 78)

Reply:
A bit surprised "Before PR" 10-thread run is much slower than 1-thread run. A quick peek into Tink AEAD implementation, it uses thread local. Unclear where the slow down comes from.

Reply:
"Musth slower" refers to latency doesn't linearly reduce with increased threads. Latency degrades quite substantially with 10x increase in concurrency.

Reply:
I am wondering why do you need this at all instead of Guice IoC?